tests/test_laserframe.py

```diff
@@ -242,6 +245,53 @@ def test_init_bad_initial_count3(self):
         with pytest.raises(ValueError, match=re.escape(f"Initial count ({initial_count}) cannot exceed capacity ({capacity}).")):
             _ = LaserFrame(capacity=capacity, initial_count=initial_count)
 
+    def test_save_and_load_snapshot(self):
+        with tempfile.NamedTemporaryFile(suffix=".h5", delete=False) as tmp:
```

Comment:
I think you want to leave `delete` `True` and set `delete_on_close = False`. Then you can put the the entire `try:` block below into the `with tempfile...` block and not need the `finally` at the end (that's the point of the `with tempfile...` construct).